service/history/handler.go

```diff
@@ -1366,7 +1366,7 @@ func (h *Handler) SyncShardStatus(ctx context.Context, request *historyservice.S
 	}
 
 	// shard ID is already provided in the request
-	engine, err := h.controller.getEngineForShard(int(request.GetShardId()))
+	engine, err := h.controller.getEngineForShard(int32(request.GetShardId()))
```

Comment:
You can change it in protos too. Just remove `int64` field completely and add new `int32 ShardId`. Let's do it as separate PR.

Reply:
SyncShardStatus should be done as part of NDC readiness. I will handle it in another PR :)